### PR TITLE
CLOUDP-66611: create datalake

### PR DESCRIPTION
**Pull request by robcarlan-mlab**
## Proposed changes

_Jira ticket:_ [mongocli datalake create](https://jira.mongodb.org/browse/CLOUDP-66611)

## Checklist

- [] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have added any necessary documentation (if appropriate)
- [] I have run `make fmt` and formatted my code

## Further comments

Create:
```
> ./bin/mongocli atlas datalake create new
{
	"cloudProviderConfig": {
		"aws": {}
	},
	"dataProcessRegion": {},
	"groupId": "5cac6a2ba6f239298a18a025",
	"hostnames": [
		"new-ta1ir.a.query.mongodb.net"
	],
	"name": "new",
	"state": "UNVERIFIED",
	"storage": {}
}
```

List:
```
> ./bin/mongocli atlas datalake ls
[
	{
		"cloudProviderConfig": {
			"aws": {
				"iamAssumedRoleARN": "arn:aws:iam::358363220050:role/data-lake-role"
			}
		},
		"dataProcessRegion": {},
		"groupId": "5cac6a2ba6f239298a18a025",
		"hostnames": [
			"datalake0-ta1ir.a.query.mongodb.net"
		],
		"name": "DataLake0",
		"state": "ACTIVE",
		"storage": {
			"databases": [
				{
					"name": "rob-data-lake",
					"collections": [
						{
							"name": "*",
							"dataSources": [
								{
									"storeName": "rob-data-lake",
									"path": "{collectionName()}"
								}
							]
						}
					]
				}
			],
			"stores": [
				{
					"name": "rob-data-lake",
					"provider": "s3",
					"region": "US_EAST_1",
					"bucket": "rob-data-lake",
					"delimiter": "/",
					"includeTags": false
				}
			]
		}
	},
	{
		"cloudProviderConfig": {
			"aws": {}
		},
		"dataProcessRegion": {},
		"groupId": "5cac6a2ba6f239298a18a025",
		"hostnames": [
			"new-ta1ir.a.query.mongodb.net"
		],
		"name": "new",
		"state": "UNVERIFIED",
		"storage": {}
	}
]

```

### CODE REVIEW

File: internal/cli/atlas/data_lake_create.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+func (opts *DataLakeCreateOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *DataLakeCreateOpts) Run() error {
+	createRequest := mongodbatlas.DataLakeCreateRequest{
+		Name: opts.name,
+	}
```
Comment (gssbzn) [inline internal/cli/atlas/data_lake_create.go:44]: [non-blocking] you could do this in the store and pass just the name to the store given that you only create by name but don't feel like you need to change this

Comment (robcarlan-mlab) [reply internal/cli/atlas/data_lake_create.go:44]: 👍  I can follow whatever we do normally, is there a preference?

Comment (gssbzn) [reply internal/cli/atlas/data_lake_create.go:44]: for single params we prefer to do it on the store, reduces how much we expose of the api to the cli

another thing we've done for more complex creates (loads of flags) is to have a `opts.newAPIType()` where we encapsulate this, but for this one so simple store or how you have works fine

Comment (robcarlan-mlab) [reply internal/cli/atlas/data_lake_create.go:44]: (I can go back to this when I look at update)

**Pull request merged**